platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/model/internal/deterministic/DeterministicHeartbeatScheduler.java

```diff
@@ -71,13 +71,13 @@ public void tick() {
         for (final Entry<Duration, List<HeartbeatTask>> entry : heartbeatsByPeriod.entrySet()) {
             final Duration period = entry.getKey();
             final List<HeartbeatTask> tasksForPeriod = entry.getValue();
-            final Instant lastHeartbeat = lastHeartbeats.get(period);
+            final Instant lastHeartbeat = previousHeartbeats.get(period);
```

Comment:
nit: local variable still called `lastHeartbeat`

feel free to leave as is, since it's not really a problem

Reply:
I had to fix a merge conflict, no extra overhead. Fixed.